src/main/java/com/gmail/goosius/siegewar/utils/TownPeacefulnessUtil.java

```diff
@@ -239,22 +213,110 @@ public static void evaluatePeacefulTownNationAssignments() {
 		}
 		//Send a global message with how many towns were modified.
 		if (modifiedTowns > 0) {
-			Boolean one = modifiedTowns == 1;
+			boolean one = modifiedTowns == 1;
 			Messaging.sendGlobalMessage(Translation.of("msg_war_siege_peaceful_town_total_switches", modifiedTowns, one ? "" : "s", one ? "has" : "have"));
 		}
 	}
 
-	public static Set<Nation> getValidGuardianNations(Town peacefulTown) {
-		Set<Town> validGuardianTowns = getValidGuardianTowns(peacefulTown);
-		Set<Nation> validGuardianNations = new HashSet<>();
-		for(Town validGuardianTown: validGuardianTowns) {
+	/**
+	 * 	Ensure the town is in the nation of one of the given guardian towns
+	 */
+	private static boolean ensurePeacefulTownIsInNationOfGuardianTown(Town peacefulTown, List<Town> eligibleGuardianTowns) throws Exception{
+
+		//If the given list is null, ensure town is nationless
+		if(eligibleGuardianTowns == null) {
+			if (!peacefulTown.hasNation()) {
+				//Already nationless
+				return false;
+			} else {
+				//Remove town from nation
+				Nation previousNation = null;
+				try {
+					previousNation = peacefulTown.getNation();
+				} catch (NotRegisteredException e) {}
+				TownyMessaging.sendPrefixedNationMessage(previousNation, Translation.of("msg_war_siege_peaceful_town_left_nation", peacefulTown.getFormattedName(), previousNation.getFormattedName()));
+				peacefulTown.removeNation();
+				peacefulTown.save();
```

Comment:
Town has already been saved inside of the removeNation().

Reply:
Comment addressed